docs/design_guide.md

```diff
@@ -0,0 +1,285 @@
+# r2b2 Library Design
```

Comment:
Great work overall - thanks!

I suggest writing it as the current policy. I.e. tenses should be present tense. Document what our practices are, not what they will be. The fact that this is a PR clarifies that it is still a draft.

Reply:
I think I got everything, but if I missed some tense changes please let me know!